kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/MobiComConversationFragment.java

```diff
@@ -4396,29 +4395,29 @@ public void processTakeOverFromBot(Context context, final Channel channel) {
         progressDialog.setCancelable(false);
         progressDialog.show();
 
-        KmService.removeMembersFromChannel(context, channel.getKey(), botIds, new ApplozicChannelRemoveMemberTask.ChannelRemoveMemberListener() {
+        KmService.removeMembersFromConversation(context, channel.getKey(), botIds, new KmRemoveMemberCallback() {
             @Override
-            public void onSuccess(String response, int index, Context context) {
+            public void onSuccess(String response, int index) {
                 GroupInfoUpdate groupInfoUpdate = new GroupInfoUpdate(channel);
                 Map<String, String> metadata = channel.getMetadata();
                 if (metadata == null) {
                     metadata = new HashMap<>();
                 }
-                metadata.put("CONVERSATION_ASSIGNEE", loggedInUserId);
+                metadata.put(Channel.CONVERSATION_ASSIGNEE, loggedInUserId);
                 groupInfoUpdate.setMetadata(metadata);
-                KmService.updateChannel(context, groupInfoUpdate, new AlChannelUpdateTask.AlChannelUpdateListener() {
+                KmService.updateConversation(ApplozicService.getContext(getContext()), groupInfoUpdate, new KmUpdateConversationTask.KmConversationUpdateListener() {
                     @Override
                     public void onSuccess(Context context) {
                         Message message = new Message();
 
-                        message.setMessage("Assigned to " + new ContactDatabase(context).getContactById(loggedInUserId).getDisplayName());
+                        message.setMessage(Utils.getString(context, R.string.km_assign_to_message) + new ContactDatabase(context).getContactById(loggedInUserId).getDisplayName());
 
                         Map<String, String> metadata = new HashMap<>();
-                        metadata.put("skipBot", "true");
-                        metadata.put("KM_ASSIGN", MobiComUserPreference.getInstance(context).getUserId());
-                        metadata.put("NO_ALERT", "true");
-                        metadata.put("BADGE_COUNT", "false");
-                        metadata.put("category", "ARCHIVE");
+                        metadata.put(KmService.KM_SKIP_BOT, Message.GroupMessageMetaData.TRUE.getValue());
+                        metadata.put(Message.KM_ASSIGN, MobiComUserPreference.getInstance(context).getUserId());
```

Comment:
@reytum Assignment is a property of group(Channel), groups are assigned to agent, messages are not assigned to agent, do you think Channel will be a better place to put this constant?
check the same with other properties as well.

Reply:
KM_ASSIGN and KM_STATUS are properties of messages as they are message level attributes. We check the same on Messages